clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -655,39 +907,82 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
+        }
 
         // join group
         client.prepareResponseFrom(joinGroupFollowerResponse(assignor, 1, "memberId", "leaderId", Errors.NONE.code()), coordinator);
 
         // sync group
-        client.prepareResponseFrom(syncGroupResponse(Arrays.asList(partition), Errors.NONE.code()), coordinator);
+        client.prepareResponseFrom(syncGroupResponse(partitions, Errors.NONE.code()), coordinator);
 
-        consumer.poll(0);
+        return coordinator;
+    }
 
-        // respond to the outstanding fetch so that we have data available on the next poll
-        client.respondFrom(fetchResponse(partition, 0, 5), node);
-        client.poll(0, time.milliseconds());
+    private AtomicBoolean prepareHeartbeatResponse(MockClient client, Node coordinator) {
+        final AtomicBoolean heartbeatReceived = new AtomicBoolean(false);
+        client.prepareResponseFrom(new MockClient.RequestMatcher() {
+            @Override
+            public boolean matches(ClientRequest request) {
+                heartbeatReceived.set(true);
+                return true;
+            }
+        }, new HeartbeatResponse(Errors.NONE.code()).toStruct(), coordinator);
+        return heartbeatReceived;
+    }
 
-        consumer.wakeup();
+    private AtomicBoolean prepareOffsetCommitResponse(MockClient client, Node coordinator, final Map<TopicPartition, Long> partitionOffsets) {
+        final AtomicBoolean commitReceived = new AtomicBoolean(true);
+        Map<TopicPartition, Short> response = new HashMap<>();
+        for (TopicPartition partition : partitionOffsets.keySet())
+            response.put(partition, Errors.NONE.code());
 
-        try {
-            consumer.poll(0);
-            fail();
-        } catch (WakeupException e) {
-        }
+        client.prepareResponseFrom(new MockClient.RequestMatcher() {
+            @Override
+            public boolean matches(ClientRequest request) {
+                OffsetCommitRequest commitRequest = new OffsetCommitRequest(request.request().body());
+                for (Map.Entry<TopicPartition, Long> partitionOffset : partitionOffsets.entrySet()) {
+                    OffsetCommitRequest.PartitionData partitionData = commitRequest.offsetData().get(partitionOffset.getKey());
+                    // verify that the expected offset has been committed
+                    if (partitionData.offset != partitionOffset.getValue()) {
+                        commitReceived.set(false);
+                        return false;
+                    }
+                }
+                return true;
+            }
+        }, offsetCommitResponse(response), coordinator);
+        return commitReceived;
+    }
 
-        // make sure the position hasn't been updated
-        assertEquals(0, consumer.position(partition));
+    private AtomicBoolean prepareOffsetCommitResponse(MockClient client, Node coordinator, final TopicPartition partition, final long offset) {
```

Comment:
Here too it seems like we can use the generic version of `prepareOffsetCommitResponse`.